# **НЭ. Задание `C1`**

[SmartLMS](https://edu.hse.ru/course/section.php?id=1060343)

**В части `C` нам предоставляют датасет, по которому нам и необходимо ответить на вопросы. На все задания части `C` - один датасет**

Для выполнения задания **`C1`** чаще всего необходимо сделать фильтрацию по 1-3 условиям и/или сделать сортировку. Теорию и синтаксис можно посмотреть в в [Хэндбуке, главе 2](https://edu.hse.ru/course/view.php?id=236211&section=2). Редко, но встречаются и задания на группировку - про нее прочитать можно в [Хэндбуке, главе 5](https://edu.hse.ru/course/view.php?id=236211&section=5). Например - найти категорию с наибольшим значением такого-то признака (группировка + сортировка), и среди объектов этой категории (фильтрация) найти объект с минимальным значением такого-то признака (сортировка)

В ответе просят указать значение какого-то объекта по какому-то признаку, либо количество объектов после фильтрации.

In [2]:
import pandas as pd

## **Вопрос 1**

Вам предстоит работать с датасетом, который содержит информацию о продаже автомобилей в Армении `train1_C.csv`:


| **Признак**  | **Описание**                                |
|--------------|---------------------------------------------|
| Car Name     | марка и модель автомобиля                   |
| Year         | год выпуска автомобиля                       |
| Region       | локация, где автомобиль был зарегистрирован |
| FuelType     | тип двигателя автомобиля                     |
| Mileage      | пробег автомобиля (км)                       |
| Price        | цена автомобиля (драмы)                      |
| Owners       | количество предыдущих владельцев            |


>*Среди автомобилей на бензиновом двигателе (**Gasoline**; `FuelType`), зарегистрированных в Армавире (**Armavir**; `Region`), найдите машину с самым ранним годом выпуска (`Year`). В ответ укажите модель и марку этого автомобиля (`Car Name`).*


In [3]:
df = pd.read_csv('data/train1_C.csv')

In [6]:
cond = (df['FuelType'] == 'Gasoline') & (df['Region'] == 'Armavir')
df_flt = df[cond]
df_flt.sort_values('Year').head(1)

Car Name  Year   Region  FuelType  Mileage  Price  Owners
7210  Mercedes-Benz W124  1991  Armavir  Gasoline   269000   4000       1

**Ответ**: `Mercedes-Benz W124`

---

## **Вопрос 2**

Вам предстоит работать с датасетом, который содержит информацию о домах в некотором штате США в 2014-2015 гг `train2_C.csv`. Датасет содержит следующие признаки: 


| **Признак**     | **Описание**                                    |
|-----------------|-------------------------------------------------|
| price           | стоимость дома ($)                              |
| bedrooms        | количество спален в доме                        |
| sqft_living     | площадь жилых помещений (фут²)                 |
| sqft_lot        | общая площадь (фут²)                            |
| waterfront      | дом на набережной (да — 1, нет — 0)           |
| grade           | оценка эксперта (от 1 до 13)                   |
| yr_built        | год постройки                                  |
| yr_renovated    | год ремонта (если ремонта не было — 0)         |



>*Среди домов с одной спальней (`bedrooms`) найдите дом с наименьшей общей площадью (`sqft_lot`). В ответ укажите год постройки дома (`yr_built`).*


In [10]:
df = pd.read_csv('data/train2_C.csv')

In [11]:
cond = df['bedrooms'] == 1
df_flt = df[cond]
df_flt.sort_values('sqft_lot').head(1)

price  bedrooms  sqft_living  sqft_lot  waterfront  grade  yr_built  \
21552  516500.0         1         1100       638           0      9      2014   

       yr_renovated  
21552             0

**Ответ**: 2014

---

## **Вопрос 3**

Вам предстоит работать с датасетом, который содержит синтетическую информацию о городской мобильности по дням `train3_C.csv`. Датасет содержит следующие признаки:


| **Признак**               | **Описание**                                               |
|----------------------------|------------------------------------------------------------|
| data                       | дата (год-месяц-день)                                      |
| public_transport_usage     | пассажиропоток общественного транспорта (чел./час)       |
| pedestrian_count           | интенсивность движения пешеходов (чел./час)              |
| temperature                | температура воздуха (градусы Цельсия)                     |
| humidity                   | относительная влажность воздуха (%)                        |
| public_transport_delay     | среднее время задержки общественного транспорта (мин.)    |
| day_of_week                | день недели                                                |
| pedestrian_incidents       | количество ДТП с участием пешеходов                        |



>*Среди пятниц и суббот (**Friday** и **Saturday**; `day_of_week`) найдите день с наибольшей влажностью (`humidity`). В ответ укажите значение пассажиропотока (`public_transport_usage`) для этого дня. Ответ округлите до двух знаков.*


In [12]:
df = pd.read_csv('data/train3_C.csv')

In [16]:
cond = df['day_of_week'].isin(['Friday', 'Saturday'])
df_flt = df[cond]
df_flt.sort_values('humidity', ascending=False).head(1)['public_transport_usage']

3114    306.25
Name: public_transport_usage, dtype: float64

**Ответ**: 306.25

---

## **Вопрос 4**

Вам предстоит работать с датасетом, который содержит информацию о треках канадской рок-группы Rush `train4_C.csv`. Датасет содержит следующие признаки:


| **Признак**        | **Описание**                                                                 |
|-------------------|----------------------------------------------------------------------------|
| song_name         | название трека                                                             |
| album_name        | название альбома                                                           |
| track_number      | номер трека на альбоме                                                     |
| popularity        | оценка популярности трека (0-100)                                         |
| danceability      | уровень танцевальности трека (0-1)                                         |
| energy            | уровень энергичности трека (0-1)                                          |
| loudness          | громкость трека (децибелы)                                                |
| mode              | настроение трека (мажорное — 1, минорное — 0)                              |
| instrumentalness  | вероятность того, что трек без вокала (чем ближе к 1 — меньше вокала)      |
| valence           | уровень позитивности трека (0-1)                                          |
| tempo             | темп трека (бит/мин.)                                                     |



>*В альбоме **Different Stages** (`album_name`) найдите название трека (`song_name`) с наибольшей по модулю громкостью (`loudness`).*


In [18]:
df = pd.read_csv('data/train4_C.csv')

In [22]:
cond = df['album_name'] == 'Different Stages'
df_flt = df[cond]
df_flt.sort_values('loudness', key=abs).tail(1)['song_name']

345    2112 (Discovery) - Live
Name: song_name, dtype: object

**Ответ**: `2112 (Discovery) - Live`

---

## **Вопрос 5**

Вам предстоит работать с датасетом, который содержит информацию о горнолыжных курортах `train5_C.csv`. Датасет содержит следующие признаки:


| **Признак**           | **Описание**                                             |
|-----------------------|----------------------------------------------------------|
| Resort                | название курорта                                         |
| Country               | страна, в которой находится курорт                       |
| HighestPoint          | самый высокий пик (метры над уровнем моря)             |
| LowestPoint           | самый низкий пик (метры над уровнем моря)              |
| DayPassPriceAdult     | цена билета на подъемник (€)                            |
| TotalSlope            | количество трасс                                        |
| NightSki              | наличие ночной подсветки трасс (есть — Yes, нет — No)  |
| TotalLifts            | количество подъемников                                  |
| LiftCapacity          | общая вместимость подъемников (чел.)                   |



>*Для страны (`Country`), где больше всего горнолыжных курортов, найдите название курорта (`Resort`), с наименьшей высотой пика по признаку `HighestPoint`.*


In [23]:
df = pd.read_csv('data/train5_C.csv')

In [27]:
df.groupby('Country')['Resort'].count().sort_values(ascending=False).head(2)

Country
Austria    89
France     83
Name: Resort, dtype: int64

In [28]:
cond = df['Country'] == 'Austria'
df_flt = df[cond]
df_flt.sort_values('HighestPoint').head(1)

Resort  Country  HighestPoint  LowestPoint  DayPassPriceAdult  \
206  Kirchdorf  Austria           780          640                 30   

     TotalSlope NightSki  TotalLifts  LiftCapacity  
206           5       No           4          3570

**Ответ**: `Kirchdorf`

---